installer/scripts/postinstall.sh

```diff
@@ -14,11 +14,65 @@
 # limitations under the License.
 
 
-# if there is no python2, switch on python3. Since we passed preinstall, we think that at least python3 is available.
-if ! /usr/bin/env python2 --version > /dev/null 2>&1; then
-  echo "Python2 not found, will use python3 binary for running the agent."
-  /usr/share/scalyr-agent-2/bin/scalyr-switch-python python3
-fi
+config_source_path=$(readlink /usr/share/scalyr-agent-2/bin/scalyr-agent-2-config)
+config_base_name=$(basename "${config_source_path}")
+
+is_python_valid() {
+  command=$1
+  version=$(/usr/bin/env "${command}" --version 2>&1 | grep -o "[0-9].[0-9]")
+  exit_code=$?
+
+  # shellcheck disable=SC2072,SC2071
+  if [[ -z "${version}" || "${exit_code}" -ne "0" ]]; then
+    return 1
+  elif [[ "$version" < "2.6" ]]; then
+    echo "Python ${version} is found but the minimum version for Python 2 is 2.6."
+    return 1
+  elif [[ "$version" > "3" && "$version" < "3.5" ]]; then
+    echo "Python ${version} is found but the minimum version for Python 3 is 3.5."
+    return 1
+  else
+    return 0
+  fi
+}
+
+
+check_python_version() {
+  # Verify that a suitable Python version is available and set up the agent symlinks
+  if is_python_valid python; then
+    # 'python' command exists
+    if [[ ${config_base_name} == "config_main.py" ]]; then
```

Comment:
Shouldn't this be `!=`?  I think you are trying to only run the switch command if the symlink currently doesn't point there.

I actually think we are trying to be too fancy here and should push this logic down into the `config_main.py` file.  I would advocate that the bash script always invokes `scalyr-switch-python` which then invokes `config_main.py`.  The `config_main.py` could make the determination if the symlink is already set correctly or not (and emit output to that effect).

If we take that approach, we could write automated tests to cover those cases for `config_main.py`.  It reduces the number of lines in the bash script (which I'm always fan of) and eliminates the need for the `config_base_name` variables.